### PR TITLE
Add ability to upload inline PDFs in blog

### PR DESCRIPTION
**Pull request by rstycedar**
feature request that enables authors to upload inline PDFs in Sanity IO

### CODE REVIEW

File: site/components/BlockContentRenderer/index.tsx [hunk 1]
```diff
@@ -101,6 +101,13 @@ const serializers: BlockContentProps["serializers"] = {
         </div>
       );
     },
+    pdf: (params: { node: { name: string; url: string } }) => {
+      return (
+        <Text t="body2" className={styles.paragraph}>
+          Download PDF <A href={`${params.node.url}?dl=`}>{params.node.name}</A>
```
Comment (0xAeterno) [inline site/components/BlockContentRenderer/index.tsx:107]: What do you think about allowing content writers to control the "Download PDF" CTA copy?
Could we see a use case where we might not want have this hardcoded?

**Review by 0xAeterno — COMMENTED** [review]
This is really cool @rstycedar, nice work 🔥🔥🔥
I learnt a lot going through this PR of yours!

Couple things I thought about while reviewing this:
- How might we apply the pdf download link to an image or button? I think this was flagged as potential use cases for this functionality too.

**Review by Atmosfearful — CHANGES_REQUESTED** [review]
Indeed me and @rstycedar spoke already but I'm posting here for transparency:
ideally this can be a generic "file" of known file types, and can just be inline text rather than a text block.
I believe sanity refers to this as 'annotation' https://www.sanity.io/docs/configuration#057f677d86d9

**Comment by rstycedar** [comment]
> Indeed me and @rstycedar spoke already but I'm posting here for transparency: ideally this can be a generic "file" of known file types, and can just be inline text rather than a text block. I believe sanity refers to this as 'annotation' https://www.sanity.io/docs/configuration#057f677d86d9

@Atmosfearful @0xAeterno we can also add a displayAsButton option to the schema if we wanted to let authors choose to display the annotations as a button

File: cms/schemas/blockContent.js [hunk 2]
```diff
@@ -48,6 +50,25 @@ export default {
...
+            fields: [
+              {
+                name: "item",
+                type: "reference",
+                to: [{ type: "fileUpload" }],
+                validation: (Rule) => Rule.required(),
+              },
+            ],
+            blockEditor: {
+              icon: () => <div>🗂</div>,
```
Comment (Atmosfearful) [inline cms/schemas/blockContent.js:70]: looks like a weird black blob on my windows  screen, can we try this emoji instead:
```suggestion
              icon: () => <div>📂</div>,
```

**Review by Atmosfearful — CHANGES_REQUESTED** [review]

File: cms/schemas/blockContent.js (hunk 2 above)
Comment (Atmosfearful) [reply cms/schemas/blockContent.js:70]: image

File: site/components/BlockContentRenderer/index.tsx [hunk 3]
```diff
@@ -82,6 +82,14 @@ const serializers: BlockContentProps["serializers"] = {
         </A>
       );
     },
+    internalLink: ({ children, mark }) => {
+      const { href } = mark;
+      return (
+        <A className={styles.link} href={`${href}?dl=`}>
```
Comment (Atmosfearful) [inline site/components/BlockContentRenderer/index.tsx:88]: lets stick with browser default i dont think most people want to save it

```suggestion
        <A className={styles.link} href={`${href}`}>
```

**Review by Atmosfearful — CHANGES_REQUESTED** [review]

File: site/components/BlockContentRenderer/index.tsx (hunk 3 above)
Comment (rstycedar) [reply site/components/BlockContentRenderer/index.tsx:88]: good call that would come off as fishy

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**